File: SSUMobile/Modules/Email/SSUEmailModule.m
//
//  SSUEmailModule.m
//  SSUMobile
//
//  Created by Eric Amorde on 9/9/15.
//  Copyright (c) 2015 Sonoma State University Department of Computer Science. All rights reserved.
//

#import "SSUEmailModule.h"
#import "SSUEmailConstants.h"
#import "SSUConfiguration.h"

@implementation SSUEmailModule

+ (instancetype) sharedInstance {
    static SSUEmailModule * instance = nil;
    static dispatch_once_t token;
    dispatch_once(&token, ^{
        instance = [[self alloc] init];
    });
    return instance;
}

#pragma mark - SSUModule

- (nonnull NSString *) title {
    return NSLocalizedString(@"Email",
                             @"Provides access to campus email accounts");
}

- (nonnull NSString *) identifier {
    return @"email";
}

- (UIView *) viewForHomeScreen {
    return [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"email_icon"]];
}

- (UIImage *) imageForHomeScreen {
    return [UIImage imageNamed:@"email_icon"];
}

- (BOOL) shouldNavigateToModule {
    if (![[SSUConfiguration sharedInstance] boolForKey:SSUEmailLoginEnabledKey]) {
        // The custom login must be broken (ex. something changed on Sonoma's website)
        // so we will show the user the webpage instead
        NSURL * loginURL = [NSURL URLWithString:[[SSUConfiguration sharedInstance] stringForKey:SSUEmailLDAPURLKey]];
        [[UIApplication sharedApplication] openURL:loginURL];
        return NO;
    }
    return YES;
}

- (UIViewController *) initialViewController {
    UIStoryboard * storyboard = [UIStoryboard storyboardWithName:@"Email"
                                                          bundle:[NSBundle bundleForClass:[self class]]];
    return [storyboard instantiateInitialViewController];
}

- (BOOL) showModuleInNavigationBar {
    return NO;
}

@end

File: SSUMobile/Modules/Email/SSUEmailModule.swift
//
//  SSUEmailModule.swift
//  SSUMobile
//
//  Created by Eric Amorde on 4/8/17.
//  Copyright © 2017 Sonoma State University Department of Computer Science. All rights reserved.
//

import Foundation

class SSUEmailModule: SSUModuleBase, SSUModuleUI {
    
    @objc(sharedInstance)
    static let instance = SSUEmailModule()
    
    // MARK: SSUModule
    
    var title: String {
        return NSLocalizedString("Email", comment: "Provides access to campus email accounts")
    }
    
    var identifier: String {
        return "email"
    }
    
    func setup() {
        
    }
    
    // MARK: SSUModuleUI
    
    func imageForHomeScreen() -> UIImage? {
        return UIImage(named: "email_icon")
    }
    
    func initialViewController() -> UIViewController {
        let storyboard = UIStoryboard(name: "Email", bundle: Bundle(for: type(of: self)))
        return storyboard.instantiateInitialViewController()!
    }
    
    func shouldNavigateToModule() -> Bool {
        if !SSUConfiguration.sharedInstance().bool(forKey: SSUEmailLoginEnabledKey) {
            // The custom login must be broken (ex. something changed on Sonoma's website)
            // so we will show the user the webpage instead
            if let urlString = SSUConfiguration.sharedInstance().string(forKey: SSUEmailLDAPURLKey),
                let url = URL(string:urlString) {
                UIApplication.shared.openURL(url)
            }
            return false
        }
        return true
    }
}
